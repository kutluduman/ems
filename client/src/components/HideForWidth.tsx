import { ReactElement } from "react";
import { Breakpoint, Theme, useMediaQuery } from "@mui/material";

export enum Width {
  xsDown = "xsDown",
  smDown = "smDown",
  mdDown = "mdDown",
  lgDown = "lgDown",
  xlDown = "xlDown",
  xsUp = "xsUp",
  smUp = "smUp",
  mdUp = "mdUp",
  lgUp = "lgUp",
  xlUp = "xlUp",
}

interface HideForWidthProps {
  width: Width;
  children?: React.ReactNode;
}

const HideForWidth = ({
  width,
  children,
}: HideForWidthProps): ReactElement<any> => {
  const breakpoint = width.substring(0, 2) as Breakpoint;

  const hiddenUp = useMediaQuery((theme: Theme) =>
    theme.breakpoints.up(breakpoint)
  );
  const hiddenDown = useMediaQuery((theme: Theme) =>
    theme.breakpoints.down(breakpoint)
  );

  if (width.includes("Down")) {
    return (hiddenDown ? null : children) as ReactElement;
  }

  if (width.includes("Up")) {
    return (hiddenUp ? null : children) as ReactElement;
  }

  return null as never as ReactElement;
}
export default HideForWidth;