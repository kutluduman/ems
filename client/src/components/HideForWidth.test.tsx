import { create } from "react-test-renderer";

import ThemeConfig from "../theme";
import HideForWidth, { Width } from "./HideForWidth";

describe("HideForWidth Component", () => {
  test("renders the HideForWidth Component", () => {
    const tree = create(
      <ThemeConfig>
        <HideForWidth width={Width.lgDown}>
          <h1>Some Content</h1>
        </HideForWidth>
      </ThemeConfig>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});