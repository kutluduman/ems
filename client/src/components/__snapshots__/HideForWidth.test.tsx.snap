// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HideForWidth Component renders the HideForWidth Component 1`] = `
<h1>
  Some Content
</h1>
`;
